<launch>
    <arg name="name" default="sems_experiment_mouse"/>
    <arg name="rate" default="1"/>
    <arg name="objectBasePath" default="$(find sems_exps)/config/sems_experiments.xml"/>    
    
    <node name="rosbag_play" pkg="rosbag" type="play" args="$(find sems_exps)/bags/$(arg name).bag -r $(arg rate)" required="true"/>
            
    <node name="republish_compressed" type="republish" pkg="image_transport" output="screen" args="compressed in:=/camera/image_raw raw out:=/camera/image_raw" />
    
    <node name="extended_object_detection" pkg="extended_object_detection" type="extended_object_detection_node" output="screen">
        
        <param name="objectBasePath" value="$(arg objectBasePath)"/>
        <param name="videoProcessUpdateRate" value="5"/>
        <param name="screenOutput" value="false"/>
        <param name="publishImage" value="true"/>
        <param name="publishMarkers" value="true"/>
        <param name="subscribeDepth" value="false"/>        
        <param name="maxContourPoints" value="-1"/>        
        
        <rosparam param="selectedOnStartSimple">[]</rosparam>
        <rosparam param="selectedOnStartComplex">[]</rosparam>
        
        <remap from="camera/image_raw" to="/camera/image_raw"/>
        <remap from="camera/info" to="/camera/camera_info"/>
    </node>
    
    <node name="mouse_experiment" pkg="sems_exps" type="mouse_experiment.py">
        <param name="save_path" value="$(find sems_exps)/share"/>
        
        <remap from="simple_objects" to="extended_object_detection/simple_objects"/>
    </node>
    
    <!-- visualization -->
    <node pkg="rviz" type="rviz" name="rviz"
    args="-d $(find extended_object_detection)/config/eod_example.rviz" />
    
    <node pkg="tf" type="static_transform_publisher" name="static_cam_pub"
        args="0 0 0 -1.57 0 -1.57 map cam_optical_link 100"/>  
    
</launch>
    
